lms/static/my.js

```diff
@@ -70,7 +133,14 @@ String.prototype.format = function(kwargs) {
 window.escapeUnicode = escapeUnicode;
 
 window.addEventListener('load', () => {
+  const codeElement = document.getElementById('code-view').dataset
+  const getLinkButton = document.getElementById('get-shared-link');
+  const solutionId = codeElement.id;
+  const shared = codeElement.shared;
+  const sharedByUser = codeElement.userShared;
   updateNotificationsBadge();
   trackReadAllNotificationsButton(document.getElementById('read-notifications'));
   trackCopyCodeButton(document.getElementById('copy-button'));
+  shareSolution(solutionId, shared, sharedByUser, document.getElementById('solution-link'), getLinkButton);
```

Comment:
This is way too much parameters. It might suggest that the API design can be sharpen a bit